astronomer/providers/amazon/aws/hooks/batch_client.py

```diff
@@ -0,0 +1,256 @@
+        'SUCCEEDED', else raise an AirflowException
+
+        :param job_id: a Batch job ID
+
+        :raises: AirflowException
+        """
+        job = await self.get_job_description(job_id)
+        job_status = job.get("status")
+        if job_status == self.SUCCESS_STATE:
+            self.log.info("AWS Batch job (%s) succeeded: %s", job_id, job)
+            return True
+
+        if job_status == self.FAILURE_STATE:
+            raise AirflowException(f"AWS Batch job ({job_id}) failed: {job}")
+
+        if job_status in self.INTERMEDIATE_STATES:
+            raise AirflowException(f"AWS Batch job ({job_id}) is not complete: {job}")
+
+        raise AirflowException(f"AWS Batch job ({job_id}) has unknown status: {job}")
+
+    @staticmethod
+    async def delay(delay: Union[int, float, None] = None) -> None:  # type: ignore[override]
+        """
+        Pause execution for ``delay`` seconds.
+
+        :param delay: a delay to pause execution using ``time.sleep(delay)``;
+            a small 1 second jitter is applied to the delay.
+
+        .. note::
+            This method uses a default random delay, i.e.
+            ``random.uniform(DEFAULT_DELAY_MIN, DEFAULT_DELAY_MAX)``;
+            using a random interval helps to avoid AWS API throttle limits
+            when many concurrent tasks request job-descriptions.
+        """
+        if delay is None:
+            """Using random.uniform() causes Bandit security check to fail with Issue: [B311:blacklist] Standard
+            pseudo-random generators are not suitable for security/cryptographic purposes. Hence,
+            using random.sample() instead"""
+            delay = sample(
+                list(range(BatchClientHookAsync.DEFAULT_DELAY_MIN, BatchClientHookAsync.DEFAULT_DELAY_MAX)), 1
+            )[0]
+        else:
+            delay = BatchClientHookAsync.add_jitter(delay)
+        await asyncio.sleep(delay)
+
+    async def wait_for_job(self, job_id: str, delay: Union[int, float, None] = None) -> None:  # type: ignore[override]
+        """
+        Wait for Batch job to complete
+
+        :param job_id: a Batch job ID
+
+        :param delay: a delay before polling for job status
+
+        :raises: AirflowException
+        """
+        await self.delay(delay)
+        await self.poll_for_job_running(job_id, delay)
+        await self.poll_for_job_complete(job_id, delay)
+        self.log.info("AWS Batch job (%s) has completed", job_id)
+
+    async def poll_for_job_complete(  # type: ignore[override]
+        self, job_id: str, delay: Union[int, float, None] = None
+    ) -> None:
+        """
+        Poll for job completion. The status that indicates job completion
+        are: 'SUCCEEDED'|'FAILED'.
+
+        So the status options that this will wait for are the transitions from:
+        'SUBMITTED'>'PENDING'>'RUNNABLE'>'STARTING'>'RUNNING'>'SUCCEEDED'|'FAILED'
+
+        :param job_id: a Batch job ID
+
```

Comment:
you can remove this extra blank line

Reply:
done in [f8c026d](https://github.com/astronomer/astronomer-providers/pull/404/commits/f8c026d9b7dccc417b17fd158282d0b265290ecf)